haystack/document_stores/pinecone.py

```diff
@@ -0,0 +1,682 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from haystack.nodes.retriever import BaseRetriever
+
+import logging
+from pathlib import Path
+from typing import Union, List, Optional, Dict, Generator
+from tqdm.auto import tqdm
+
+import pinecone
+import numpy as np
+
+from haystack.schema import Document
+from haystack.document_stores.sql import SQLDocumentStore
+from haystack.document_stores.base import get_batches_from_generator
+from inspect import Signature
+
+logger = logging.getLogger(__name__)
+
+
+class PineconeDocumentStore(SQLDocumentStore):
+    """
+    Document store for very large scale embedding based dense retrievers like the DPR.
+
+    It implements the Pinecone vector database (https://www.pinecone.io)
+    to perform similarity search on vectors.
+
+    The document text is stored using the SQLDocumentStore, while
+    the vector embeddings and metadata (for filtering) are indexed in a Pinecone Index.
+    """
+
+    top_k_limit = 10_000
+    top_k_limit_vectors = 1_000
+
+    def __init__(
+        self,
+        api_key: str,
+        environment: str = "us-west1-gcp",
+        sql_url: str = "sqlite:///pinecone_document_store.db",
+        pinecone_index: Optional["pinecone.Index"] = None,
+        vector_dim: int = 768,
```

Comment:
We deprecated `vector_dim` for FAISS and Milvus and use `embedding_dim` instead.

Reply:
updated